File: docs/identities/get-started/_common/code-examples/js/session/check-session.js
export const createRequireAuth = (ory, baseUrl = process.env.ORY_SDK_URL) => {
  return async (req, res, next) => {
    try {
      const { data: session } = await ory.toSession({
        cookie: req.header("cookie"),
      })
      req.session = session
      next()
    } catch (error) {
      res.redirect(`${baseUrl}/self-service/login/browser`)
    }
  }
}

export const registerSessionRoute = (app, requireAuth) => {
  app.get("/session", requireAuth, (req, res) => {
    res.json(req.session.identity.traits) // { email: 'newtestuser@gmail.com' }
  })
}

File: docs/identities/get-started/_common/code-examples/go/session/check-session.go
package main

import (
	"context"
	"fmt"
	"net/http"
	
	ory "github.com/ory/client-go"
)

func authMiddleware(next http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		// Initialize the Ory client
		configuration := ory.NewConfiguration()
		configuration.Servers = []ory.ServerConfiguration{
			{
				URL: "https://$PROJECT_SLUG.projects.oryapis.com",
			},
		}
		client := ory.NewAPIClient(configuration)
		
		// Check if user is authenticated
		cookie := r.Header.Get("Cookie")
		session, _, err := client.FrontendApi.ToSession(context.Background()).Cookie(cookie).Execute()
		
		if err != nil {
			// Not authenticated, redirect to login
			http.Redirect(w, r, "/login", http.StatusFound)
			return
		}
		
		// User is authenticated, add session to context
		ctx := context.WithValue(r.Context(), "session", session)
		next.ServeHTTP(w, r.WithContext(ctx))
	})
}

func dashboardHandler(w http.ResponseWriter, r *http.Request) {
	// Get session from context
	session := r.Context().Value("session").(*ory.Session)
	
	// Access user data
	userId := session.Identity.Id
	email := session.Identity.Traits["email"]
	
	// Render dashboard with user data
	fmt.Fprintf(w, "Welcome, %s", email)
}